pkg/inject/inject.go

```diff
@@ -545,12 +559,6 @@ func (conf *ResourceConfig) injectPodSpec(values *patch) {
 	values.Global.IdentityTrustDomain = idctx.GetTrustDomain()
 	values.Identity = &l5dcharts.Identity{}
 
-	values.AddRootVolumes = len(conf.pod.spec.Volumes) == 0
-
```

Comment:
This code is moved up, as there is a return path in the middle i.e when identity is disabled (Trace should be able to be added even when identity is disabled)